Doc entries for S3 server side encryption with customer key (SSE-C) (#24186)
trinodb/trino: docs/src/main/sphinx/object-storage/file-system-s3.md

* - `s3.sse.type`

>   - Set the type of S3 server-side encryption (SSE) to use. Defaults to `NONE`

Reviewer: Please add to the description of the PR that it relates to https://github.com/trinodb/trino/pull/22992

    for no encryption. Other valid values are `S3` for encryption by S3 managed
    keys, `KMS` for encryption with a key from the AWS Key Management

>     Service (KMS), and `CUSTOMER` for encryption with customer-provided key. Note that S3 automatically uses SSE so `NONE` and `S3` are

Reviewer 2: add that the key is provided with s3.sse.customer-key

>     Service (KMS), and `CUSTOMER` for encryption with customer-provided key within `s3.sse.customer-key`. Note that S3 automatically uses SSE so `NONE` and `S3` are

Reviewer 2: ```suggestion
    Service (KMS), and `CUSTOMER` for encryption with a customer-provided key from `s3.sse.customer-key`. Note that S3 automatically uses SSE so `NONE` and `S3` are
```

    equivalent. S3-compatible systems might behave differently.
* - `s3.sse.kms-key-id`
  - The identifier of a key in KMS to use for SSE.
* - `s3.sse.customer-key`

>   - Customer Key to use in SSE-C mode for SSE.

Reviewer 2: Why the capitalization "Customer Key" .. is that something specifically from AWS .. can we link to the docs for that. Or otherwise can we add more info on what this property needs .. is it a path to a file or an actual value? Can we add an example?

>   - This is a 256-bit, base64-encoded encryption key (using encryption algorithm AES-256) to be used for Amazon S3 to encrypt or decrypt data within SSE-C mode for SSE (`s3.sse.type`=`CUSTOMER`) 

Reviewer 2: ```suggestion
  - The 256-bit, base64-encoded AES-256 encryption key to to encrypt or decrypt data from S3 when using the SSE-C mode for SSE wiht `s3.sse.type`=`CUSTOMER`.
```
